operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakAdminSecret.java

```diff
@@ -0,0 +1,57 @@
+package org.keycloak.operator.v2alpha1;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.utils.KubernetesResourceUtil;
+import org.keycloak.operator.OperatorManagedResource;
+import org.keycloak.operator.v2alpha1.crds.Keycloak;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.UUID;
+
+public class KeycloakAdminSecret extends OperatorManagedResource {
+
+    private final Keycloak kc;
+    private final String secretName;
+    private final Base64.Encoder base64Encoder = Base64.getEncoder();
+
+    public KeycloakAdminSecret(KubernetesClient client, Keycloak keycloak) {
+        super(client, keycloak);
+        this.kc = keycloak;
+        this.secretName = KubernetesResourceUtil.sanitizeName(getName() + "-admin");
+    }
+
+    @Override
+    protected Optional<HasMetadata> getReconciledResource() {
+        if (client.secrets().inNamespace(getNamespace()).withName(secretName).get() != null) {
+            return Optional.empty();
+        } else {
+            return Optional.of(createSecret());
+        }
+    }
+
+    private String encode(String value) {
+        return base64Encoder.encodeToString(value.getBytes(StandardCharsets.UTF_8));
+    }
+
+    private Secret createSecret() {
+        return new SecretBuilder()
+                .withNewMetadata()
+                .withName(secretName)
+                .withNamespace(getNamespace())
+                .endMetadata()
+                .withType("kubernetes.io/basic-auth")
+                .addToData("username", encode("admin"))
+                .addToData("password", encode(UUID.randomUUID().toString().replace("-", "")))
```

Comment:
```suggestion
                .addToStringData("password", UUID.randomUUID().toString().replace("-", ""))
```

Reply:
I guess it's ok, testing it now (not merging your suggestions since those things needs follow up modifications but I'm incorporating the feedback)